### ISSUE TITLE
Worker node loses labels after VM reboot

### ISSUE THREAD

**Issue by lwouis** [rancher/rke#1054]
**RKE version:**
```
$ rke -v
rke version v0.1.11
```

**Docker version: (`docker version`,`docker info` preferred)**
```
$ docker version
Client:
 Version:      17.03.2-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 01:35:00 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.2-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 01:35:00 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
vSphere (10 VMs with Ubuntu, hypervised by VMWare ESXi)

**cluster.yml file:**
```
# If you intended to deploy Kubernetes in an air-gapped environment,
# please consult the documentation on how to configure custom RKE images.
nodes:
- address: 192.168.11.11
  role:
  - controlplane
  - etcd
  user: <REDACTED>
- address: 192.168.11.12
  role:
  - worker
  user: <REDACTED>
  labels:
    agentpool: db
- address: 192.168.11.13
  role:
  - worker
  user: <REDACTED>
  labels:
    agentpool: services
- address: 192.168.11.14
  role:
  - worker
  user: <REDACTED>
  labels:
    agentpool: services
- address: 192.168.11.15
  role:
  - worker
  user: <REDACTED>
  labels:
    agentpool: services
- address: 192.168.11.16
  role:
  - worker
  user: <REDACTED>
  labels:
    agentpool: services
- address: 192.168.11.17
  role:
  - worker
  user: <REDACTED>
  labels:
    agentpool: services
- address: 192.168.11.18
  role:
  - worker
  user: <REDACTED>
  labels:
    agentpool: services
- address: 192.168.11.19
  role:
  - worker
  user: <REDACTED>
  labels:
    agentpool: services
- address: 192.168.11.20
  role:
  - worker
  user: <REDACTED>
  labels:
    agentpool: services
ssh_key_path: <REDACTED>
ssh_agent_auth: true
bastion_host:
  address: "<REDACTED>"
  user: "<REDACTED>"
  ssh_key_path: "<REDACTED>"
services:
  etcd:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    external_urls: []
    ca_cert: ""
    cert: ""
    key: ""
    path: ""
    snapshot: false
    retention: ""
    creation: ""
  kube-api:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    service_cluster_ip_range: 10.43.0.0/16
    service_node_port_range: ""
    pod_security_policy: false
  kube-controller:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    cluster_cidr: 10.42.0.0/16
    service_cluster_ip_range: 10.43.0.0/16
  scheduler:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
  kubelet:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
    cluster_domain: cluster.local
    infra_container_image: ""
    cluster_dns_server: 10.43.0.10
    fail_swap_on: false
  kubeproxy:
    image: ""
    extra_args: {}
    extra_binds: []
    extra_env: []
network:
  plugin: canal
  options: {}
authentication:
  strategy: x509
  options: {}
  sans: []
addons: ""
addons_include: []
authorization:
  mode: rbac
  options: {}
ignore_docker_version: false
system_images:
  etcd: rancher/coreos-etcd:v3.2.18
  alpine: rancher/rke-tools:v0.1.15
  nginx_proxy: rancher/rke-tools:v0.1.15
  cert_downloader: rancher/rke-tools:v0.1.15
  kubernetes_services_sidecar: rancher/rke-tools:v0.1.15
  kubedns: rancher/k8s-dns-kube-dns-amd64:1.14.10
  dnsmasq: rancher/k8s-dns-dnsmasq-nanny-amd64:1.14.10
  kubedns_sidecar: rancher/k8s-dns-sidecar-amd64:1.14.10
  kubedns_autoscaler: rancher/cluster-proportional-autoscaler-amd64:1.0.0
  kubernetes: rancher/hyperkube:v1.11.3-rancher1
  flannel: rancher/coreos-flannel:v0.10.0
  flannel_cni: rancher/coreos-flannel-cni:v0.3.0
  calico_node: rancher/calico-node:v3.1.3
  calico_cni: rancher/calico-cni:v3.1.3
  calico_controllers: ""
  calico_ctl: rancher/calico-ctl:v2.0.0
  canal_node: rancher/calico-node:v3.1.3
  canal_cni: rancher/calico-cni:v3.1.3
  canal_flannel: rancher/coreos-flannel:v0.10.0
  wave_node: weaveworks/weave-kube:2.1.2
  weave_cni: weaveworks/weave-npc:2.1.2
  pod_infra_container: rancher/pause-amd64:3.1
  ingress: rancher/nginx-ingress-controller:0.16.2-rancher1
  ingress_backend: rancher/nginx-ingress-controller-defaultbackend:1.4
  metrics_server: rancher/metrics-server-amd64:v0.2.1
ingress:
  provider: ""
  options: {}
  node_selector: {}
  extra_args: {}
cluster_name: ""
cloud_provider:
  name: "vsphere"
  vsphereCloudProvider:
    global:
      insecure-flag: true
    virtual_center:
      <REDACTED>:
        user: <REDACTED>
        password: <REDACTED>
        datacenters: ha-datacenter
    workspace:
      server: <REDACTED>
      datacenter: <REDACTED>
      folder: kubernetes
      default-datastore: <REDACTED>
      resourcepool-path: Resources
private_registries:
- url: <REDACTED>
  user: <REDACTED>
  password: <REDACTED>
prefix_path: ""
addon_job_timeout: 0
monitoring:
  provider: ""
  options: {}
```

**Steps to Reproduce:**

1. Initial state. We observe that uvm06 is listed, and has proper agentpool=services and node-role.kubernetes.io/worker=true labels, as well as the role worker.

```
$ kc get nodes --show-labels
NAME    STATUS   ROLES               AGE   VERSION   LABELS
uvm01   Ready    controlplane,etcd   15d   v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.11,node-role.kubernetes.io/controlplane=true,node-role.kubernetes.io/etcd=true
uvm02   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.12
uvm03   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.13
uvm04   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.14
uvm05   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.15
uvm06   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.16,node-role.kubernetes.io/worker=true
uvm07   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.17,node-role.kubernetes.io/worker=true
uvm08   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.18,node-role.kubernetes.io/worker=true
uvm09   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.19,node-role.kubernetes.io/worker=true
uvm10   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.20,node-role.kubernetes.io/worker=true

2. Go to ESXi web UI > VM06 > press the "Shut down" button. Wait until the VM is shut down. We now observe that node uvm06 is no longer listed

$ kc get nodes --show-labels
NAME    STATUS   ROLES               AGE   VERSION   LABELS
uvm01   Ready    controlplane,etcd   15d   v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.11,node-role.kubernetes.io/controlplane=true,node-role.kubernetes.io/etcd=true
uvm02   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.12
uvm03   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.13
uvm04   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.14
uvm05   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.15
uvm07   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.17,node-role.kubernetes.io/worker=true
uvm08   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.18,node-role.kubernetes.io/worker=true
uvm09   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.19,node-role.kubernetes.io/worker=true
uvm10   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.20,node-role.kubernetes.io/worker=true
```

3. Press the "Start" button. Wait until the VM is up. We now observe that node uvm06 is back in the list, but it lost its agentpool=services and node-role.kubernetes.io/worker=true labels, as well as the role worker.

```
$ kc get nodes --show-labels
NAME    STATUS   ROLES               AGE   VERSION   LABELS
uvm01   Ready    controlplane,etcd   15d   v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.11,node-role.kubernetes.io/controlplane=true,node-role.kubernetes.io/etcd=true
uvm02   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.12
uvm03   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.13
uvm04   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.14
uvm05   Ready    <none>              1d    v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.15
uvm06   Ready    <none>              13s   v1.11.3   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.16
uvm07   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.17,node-role.kubernetes.io/worker=true
uvm08   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.18,node-role.kubernetes.io/worker=true
uvm09   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.19,node-role.kubernetes.io/worker=true
uvm10   Ready    worker              15d   v1.11.3   agentpool=services,beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/hostname=192.168.11.20,node-role.kubernetes.io/worker=true
```

**Results:**
We observe that after rebooting the node uvm06 through ESXi web UI, once it is back up, it lost its agentpool=services and node-role.kubernetes.io/worker=true labels, as well as the role worker.

Interestingly, I rebooted the master node, and it didn't affect it. It still had the controlplane,etcd roles, and proper labels after reboot.

**Comment by superseb** [comment]
Can you try it on latest release please? I think this is fixed in v1.11.5: https://github.com/kubernetes/kubernetes/pull/70291

**Comment by superseb** [comment]
Let me know if it doesn't solve the issue.
